### DISCUSSION TITLE
FOTA (over the air) image format

### DISCUSSION THREAD

**Discussion by projecta98** [mcu-tools/mcuboot#1561] [General]
Hi there,
I am planning to use mcuboot for a new project based on PSOC6.
Are there any examples for creating an upgrade image for over the air update (i.e. converting the HEX file to some binary format)?
Or Is the MCU supposed to be able to parse the generated .HEX file (while writing it to external flash)?
 
My current application size is only 300KB while the maximum image size is set to 1.9MB, so creating a binary image becomes 1.9MB as well (with majority of the image filled with 0xFF).  I would prefer to only send the necessary bytes over the air.

If I choose to use the smaller image size (in the MCUboot configuration), it appears that will limit the size of the future upgrades (since the MCUBoot does to recognize the larger image in the external flash for upgrade).
It appears that the image size is fixed in the bootloader, instead of picking up the image size from the **UPGRADE** image in the secondary slot (during the upgrade process).

Am I missing something obvious?

Thanks.

**Comment by de-nordic** [comment]
How are you converting the image?

**Comment by Olstyle** [comment]
The hash and signature (generated by the imgtool, which does have its own documentation) include the 0xFF values, so a complete binary image can not do without them. To have a sparse image you need to also supply the address information for every chunk of the data, which means you no longer have a binary but a hex format(or something comparable). 
What you can do ist to use something like srec_cat to remove the "empty" data and write the result as a hex file, then provide that to your application, and in the end parse that in the application to actual write commands.
